app/templates/register.html

```diff
@@ -4,30 +4,37 @@
 <h1 class="h-12 mt-8 text-3xl font-bold text-center text-green-700">
   Welcome to Financial Forecast
 </h1>
-<form action="/register" method="post" class="flex flex-col items-center justify-center w-screen h-full my-8">
+<!-- TODO: instead of reloading whole page on bad registration, reload just the form html using htmx -->
+<form action="/auth/register" method="post" class="flex flex-col items-center justify-center w-screen h-full my-8">
   {% if popup_text and username %}
-  <div
-    class="flex flex-row justify-between w-1/2 h-16 p-2 m-2 align-middle border rounded-md md:w-1/3 bg-slate-200 border-slate-400">
-    <p class="self-center w-1/2 text-4xl text-center text-yellow-500 md:w-1/3">&#9888</p>
-    <p class="self-center w-full text-sm italic text-left text-black">{{popup_text}}</p>
-  </div>
+  {% include 'fragments/invalid-popup.html' %}
   {% endif %}
+
+  <!-- red outline for invalid username -->
   {% if status == "username_exists" %}
```

Comment:
change this to have a single response status for 'invalid username/password' to prevent confirming that a particular username exists.